### PR TITLE
Pluggable Character UI

### PR DESCRIPTION
**Pull request by ChadNedzlek**
Added some declarative attributes to control plugin/view injection.

Created a Showcase project that just loads all the plugins and creates fake characters.

Menu view:
image

Sad little pathfinder view:
image

### CODE REVIEW

**Comment by adiaaida** [comment]
Can you add a screencap to the description so I can compare the Mac view?

File: GameModes/Pathfinder/Pathfinder.csproj [hunk 1]
```diff
@@ -1,4 +1,26 @@
...
+    <Compile Update="**\*.axaml.cs">
+      <DependentUpon>%(Filename)</DependentUpon>
+    </Compile>
+    <AvaloniaResource Include="**\*.axaml">
+      <SubType>Designer</SubType>
+    </AvaloniaResource>
   </ItemGroup>
-</Project>
+  <ItemGroup>
+    <AvaloniaResource Remove="Views\PathfinderCharacterView.axaml" />
```
Comment (alexperovich) [inline GameModes/Pathfinder/Pathfinder.csproj:11]: Is this intentionally removing the view we just added?

File: GameModes/Pathfinder/Pathfinder.csproj [hunk 2]
```diff
@@ -1,4 +1,26 @@
...
+  </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include="Avalonia">
```
Comment (alexperovich) [inline GameModes/Pathfinder/Pathfinder.csproj:14]: Can you include these 2 package references and the ProjectReference in the Directory.Build.props for the GameModes folder? Every plugin will need them so i'd like to keep them in the same place.

File: PluginShowcase/.gitignore [hunk 3]
```diff
@@ -0,0 +1,337 @@
+## Ignore Visual Studio temporary files, build results, and
```
Comment (alexperovich) [inline PluginShowcase/.gitignore:1]: Why a new .gitignore?

File: PluginShowcase/nuget.config [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <packageSources>
+    <add key="AvaloniaCI" value="https://www.myget.org/F/avalonia-ci/api/v2" />
```
Comment (alexperovich) [inline PluginShowcase/nuget.config:4]: This is an odd thing to add.

File: GameModes/Pathfinder/Pathfinder.csproj (hunk 1 above)
Comment (ChadNedzlek) [reply GameModes/Pathfinder/Pathfinder.csproj:11]: When you do the "dotnet new avalonia.usercontrol" this is what it does.  If you take the remove away, you get really weird errors.

File: GameModes/Pathfinder/Pathfinder.csproj (hunk 2 above)
Comment (ChadNedzlek) [reply GameModes/Pathfinder/Pathfinder.csproj:14]: Yeah, I was thinking that too.  Have a sort of... plugin starter pack of references.

File: PluginShowcase/.gitignore (hunk 3 above)
Comment (ChadNedzlek) [reply PluginShowcase/.gitignore:1]: dotnet avalonia.mvvm does this.

File: PluginShowcase/nuget.config (hunk 4 above)
Comment (ChadNedzlek) [reply PluginShowcase/nuget.config:4]: That is... odd... I'm going to remove it and make sure stuff still works.

File: PluginShowcase/.gitignore (hunk 3 above)
Comment (ChadNedzlek) [reply PluginShowcase/.gitignore:1]: I can remove it if we want, but in general I don't fight with tools much. :-)

File: PluginShowcase/nuget.config (hunk 4 above)
Comment (ChadNedzlek) [reply PluginShowcase/nuget.config]: Restore worked fine. Removed.

File: GameModes/Pathfinder/Pathfinder.csproj (hunk 2 above)
Comment (ChadNedzlek) [reply GameModes/Pathfinder/Pathfinder.csproj:14]: Done!

**Comment by ChadNedzlek** [comment]
@adiaaida It's a really, really sad little view right now.  It took me hours to try to figure out how to load a xaml dynamically.

File: GameModes/Directory.Build.props [hunk 5]
```diff
@@ -1,11 +1,31 @@
...
-
-  <ItemGroup>
+  <ItemGroup Condition=" !$(MSBuildProjectName.EndsWith('.Tests')) ">
+    <PackageReference Include="Avalonia">
+      <ExcludeAssets>runtime</ExcludeAssets>
+    </PackageReference>
+    <PackageReference Include="ReactiveUI">
+      <ExcludeAssets>runtime</ExcludeAssets>
+    </PackageReference>
+    <ProjectReference Include="$(MSBuildThisFileDirectory)..\Core\Core.csproj">
```
Comment (alexperovich) [inline GameModes/Directory.Build.props:11]: This is now referencing Core.csproj twice. I'm not really sure what that does.

File: GameModes/Directory.Build.props [hunk 6]
```diff
@@ -1,11 +1,31 @@
...
+    </ProjectReference>
     <ProjectReference Include="$(MSBuildThisFileDirectory)..\Core\Core.csproj">
       <Private>false</Private>
       <ExcludeAssets>runtime</ExcludeAssets>
     </ProjectReference>
   </ItemGroup>
+  <ItemGroup Condition=" $(MSBuildProjectName.EndsWith('.Tests')) ">
+    <PackageReference Include="Avalonia" />
+    <PackageReference Include="ReactiveUI" />
+    <ProjectReference Include="$(MSBuildThisFileDirectory)..\Core\Core.csproj" />
```
Comment (alexperovich) [inline GameModes/Directory.Build.props:27]: This one is too.

File: GameModes/Directory.Build.props (hunk 5 above)
Comment (ChadNedzlek) [reply GameModes/Directory.Build.props:11]: Whoops. Fixed both of these. :-)

File: PluginShowcase/.gitignore (hunk 3 above)
Comment (ChadNedzlek) [reply PluginShowcase/.gitignore:1]: I'll remove them all and see if anything new shows up in status.

File: GameModes/Pathfinder/Pathfinder.csproj (hunk 1 above)
Comment (ChadNedzlek) [reply GameModes/Pathfinder/Pathfinder.csproj:11]: I just removed the original add.  I'm not sure why it adds all of them and then deletes them.  But it seems to work if you don't have _any_ AvaloniaResources?

Comment (ChadNedzlek) [reply GameModes/Pathfinder/Pathfinder.csproj]: *shrugs* Oh well, I'm super confused, but it all seems to work, even after a git clean without this... so... gone?

**Comment by adiaaida** [comment]
@ChadNedzlek it looks like that on the mac too, so that's good at least :).

**Pull request merged**